### ISSUE TITLE
Temporary Services with Brigade

### ISSUE THREAD

**Issue by philmay** [Azure/brigade#459]
For some of the Brigade scripts we are running, we'd like to temporarily stand up supporting services that are available to our Brigade Jobs while our script is running. Once our Jobs have completed, we'd like to stop these supporting services as part of the cleanup process. Is this something that fits the Brigade approach to things (i.e., something Brigade might support), or would this be better handled elsewhere?

What we do now is use Helm to deploy these services and just leave them running, but some of our Brigade triggers are infrequent, and we'd rather not have unused services sitting around idle. Another option we've been kicking around is putting this service initiation into our custom Brigade gateway code – so in response to one of our external triggers, we'd deploy supporting services programmatically, then run Brigade, and then take the services down (haven't dug into the details on this yet).

Any thoughts?

Thanks!

**Comment by philmay** [comment]
Thanks for this advice, it's really helpful! 

I was discussing option 4 on Friday with one of my co-workers (we already have a custom gateway running), and we came to the conclusion that we could do it, but that it was a bit complicated, and that it would be duplicating work that Brigade is already doing. We also discussed option 1, which seemed to be the simplest approach, so I started down that path. I should have option 1 working in short order.

Option 3 sounds interesting, and I may look into that as a possible next step after I get option 1 working with our system. I'll let you know what we do and our degree of success.

My co-worker also brought up an idea similar to option 2, but having the services be sidecars to our Brigade Job pods (the ones that actually need the services) rather than the worker. The idea of attaching them to the worker sounds interesting, so I'll bring that up with the team on Monday. I may reach out to @blimmer if we decide to look into it.
